storage/encoding/ssz.go

```diff
@@ -23,23 +23,28 @@ package encoding
 import (
 	"github.com/berachain/beacon-kit/primitives/common"
 	"github.com/berachain/beacon-kit/primitives/constraints"
+	"github.com/berachain/beacon-kit/primitives/decoder"
 	"github.com/davecgh/go-spew/spew"
+	"github.com/karalabe/ssz"
 )
 
 // SSZValueCodec provides methods to encode and decode SSZ values.
 type SSZValueCodec[T interface {
-	constraints.SSZMarshallable[T]
-}] struct{}
+	ssz.StaticObject
+	constraints.SSZMarshallable
+}] struct {
+	NewEmptyF func() T // constructor
```

Comment:
we pass constructors (and generic ones) to the codec to give it the template to decode object